### PR TITLE
Fix Sequel deprecation of globally loading pg extensions

### PR DESCRIPTION
**Pull request by tomgilligan**

### CODE REVIEW

**Comment by stevehodgkiss** [comment]
We'll need to ensure the event store db connection has this applied also. I think the specs are likely passing because they share the same connection. Maybe we can do this in the config block where the 2 database connections are passed in?

**Comment by tomgilligan** [comment]
> We'll need to ensure the event store db connection has this applied also.
Ahh! Missed this >.<
> Maybe we can do this in the config block where the 2 database connections are passed in?
As in, in the apps?

----------------------------------

I was going to rename `@pg_connection` etc. to just `@connnection` https://github.com/envato/event_sourcery-postgres/blob/master/lib/event_sourcery/postgres/event_store.rb#L11
but then I found this TODO:
https://github.com/envato/event_sourcery-postgres/blob/master/spec/support/db_helpers.rb#L8
So all instances of `connection` that are a DB connection should be renamed to `pg_connection`?  I'd like to make `lib/event_sourcery/postgres/event_store.rb` match `lib/event_sourcery/postgres/tracker.rb` in how it refers to DB connection (or vice versa) as I feel like this is partly how I missed event store DB connection also needed this extension change.

Also wondering if event_tracker should be just attr_reader because it isn't just getting set by event_tracker= https://github.com/envato/event_sourcery-postgres/blob/master/lib/event_sourcery/postgres/config.rb#L43

**Comment by stevehodgkiss** [comment]
> As in, in the apps?

I was thinking in `Config#event_store_database=` and `Config#projections_database=` but maybe calling it in the EventStore class itself is better. The Tracker connection doesn't need json support but the projections might, depending if the app uses it.

> but then I found this TODO

That's mainly about making the specs consistent terms of what the connection variable is called. Happy to go with `connection`.

**Comment by tomgilligan** [comment]
> That's mainly about making the specs consistent terms of what the connection variable is called. Happy to go with connection

Decided to do this in a separate PR, settled on db_connection

File: lib/event_sourcery/postgres/config.rb [hunk 1]
```diff
@@ -38,10 +38,26 @@ def event_sink
...
         @projections_database = sequel_connection
         @event_tracker = Postgres::Tracker.new(sequel_connection)
       end
+
+      private
+
+      def setup_connection(sequel_connection)
+         return unless sequel_connection
+
+         sequel_connection.extension :pg_json
```
Comment (stevehodgkiss) [inline lib/event_sourcery/postgres/config.rb]: Too much indentation

**Pull request merged**